packages/react/src/pages/analytics-dashboard/analytics-dashboard.tsx

```diff
@@ -0,0 +1,143 @@
+import React, { useCallback, useEffect, useState } from 'react';
+import Toolbar, { Item } from 'devextreme-react/toolbar';
+import Button from 'devextreme-react/button';
+import { getOpportunitiesByCategory, getSalesByCategory, getSales, getSalesByStateAndCity, getSalesByState } from 'dx-rwa-data';
+import './analytics-dashboard.scss';
+import { RevenueSnapshotCard } from './cards/RevenueSnapshotCard';
+import { RevenueAnalysisCard } from './cards/RevenueAnalysisCard';
+import { ConversionFunnelCard } from './cards/ConversionFunnelCard';
+import { RevenueCard } from './cards/RevenueCard';
+import { ConversionTicker, LeadsTicker, OpportunitiesTicker, RevenueTotalTicker } from './cards/TickerCards';
+
+const tabItems = {
+  Week: {
+    period: '2020-01-24/2020-01-31',
+    index: 0,
+  },
+  '2 Weeks': {
+    period: '2020-01-14/2020-01-31',
+    index: 1,
+  },
+  Month: {
+    period: '2020-01-01/2020-02-01',
+    index: 2,
+  },
+  Year: {
+    period: '2020-01-01/2021-01-01',
+    index: 3,
+  },
+  All: {
+    period: '2018-01-01/2022-01-01',
+    index: 4,
+  },
+};
+
+const DEFAULT_TAB_KEY = 'All';
+
+const calculateTotal = (data: Array<any>) => {
+  if (!data) return;
+  return data.reduce((total, item) => total + (item.value || item.total), 0);
+};
+
+const DashboardCardsGroup = ({ children, compact = false }) => (
+  <div className={`cards${(compact && ' compact') || ''}`}>{compact ? React.Children.map(children, (child) => React.cloneElement(child, { compact })) : children}</div>
```

Comment:
It is difficult to read and understand. Maybe make follow changes?
1. className={`cards${(compact && ' compact') || ''}`} -> className={`cards${compact ? ' compact' : ''}`}
2. Block inside the `div` move to new line and each block after ? or : on new line too

Reply:
Rewritten this component in a more correct way